pointpats/spacetime.py

```diff
@@ -1195,42 +1208,86 @@ def statistic_(self):
 
     @classmethod
     def from_dataframe(
-        cls, dataframe, time_col, delta, tau, permutations=99, keep=False
+        cls,
+        dataframe,
+        time_col: str,
+        delta: int,
+        tau: int,
+        permutations: int = 99,
+        keep: bool = False,
     ):
-        """Compute a local Knox statistic from a dataframe of Point observations
+        """Compute a set of local Knox statistics from a dataframe of Point observations
 
         Parameters
         ----------
         dataframe : geopandas.GeoDataFrame
             dataframe holding observations. Should be in a projected coordinate system
             with geometries stored as Points
         time_col : str
-            column in the dataframe storing the timestamp for each observation
+            column in the dataframe storing the time values (integer coordinate)
+            for each observation. For example if the observations are stored with
+            a timestamp, the time_col should be converted to a series of integers
+            representing, e.g. hours, days, seconds, etc.
         delta : int
-            delta parameter defining the spatial neighbor threshold
+            delta parameter defining the spatial neighbor threshold measured in the
+            same units as the dataframe CRS
         tau : int
-            tau parameter defining the temporal neihgbor threshold
+            tau parameter defining the temporal neihgbor threshold (in the units
+            measured by `time_col`)
         permutations : int, optional
             permutations to use for computation inference, by default 99
+        keep : bool
+            whether to store realized values of the statistic under permutations
 
         Returns
         -------
-        pointpats.spacetime.Knox
-            a fitted Knox class
+        pointpats.spacetime.LocalKnox
+            a fitted KnoxLocal class
+
         """
         s_coords, t_coords = _spacetime_points_to_arrays(dataframe, time_col)
 
         return cls(s_coords, t_coords, delta, tau, permutations, keep)
 
 
 def _spacetime_points_to_arrays(dataframe, time_col):
+    """convert long-form geodataframe into arrays for kdtree
+
+    Parameters
+    ----------
+    dataframe : geopandas.GeoDataFrame
+        geodataframe with point geometries
+    time_col : str
+        name of the column on dataframe that stores time values
+
+    Returns
+    -------
+    tuple
+        two numpy arrays holding spatial coodinates s_coords (n,2)
+        and temporal coordinates t_coords (n,1)
+
+    """
+    if dataframe.crs is None:
+        warn(
+            "There is no CRS set on the dataframe. The KDTree will assume coordinates "
+            "are stored in Euclidean distances"
+        )
+    else:
+        assert (
+            dataframe.crs.is_projected
+        ), ("The input dataframe must be in a projected coordinate system, but it is "
+    f"currently set to {dataframe.crs}")
```

Comment:
```suggestion
        if dataframe.crs.is_geographic:
            raise ValueError(
                "The input dataframe must be in a projected coordinate system."
            )
```

ValueError is a better option than AssertionError. Also, printing crs like this can in some cases result in a looong string.